pkg/dbal/Tests/DbalConnectionFactoryConfigTest.php

```diff
@@ -101,6 +101,54 @@ public static function provideConfigs()
             ],
         ];
 
+        yield [
+            [
+                'dsn' => 'mysql+pdo:',
+                'connection' => [
+                    'dbname' => 'dbname',
+                ],
+            ],
+            [
+                'connection' => [
+                    'dbname' => 'dbname',
+                    'driver' => 'pdo_mysql',
+                    'host' => 'localhost',
+                    'port' => '3306',
+                    'user' => 'root',
+                    'password' => '',
+                ],
+                'table_name' => 'enqueue',
+                'polling_interval' => 1000,
+                'lazy' => true,
+            ],
+        ];
+
+        yield [
+            [
+                'dsn' => 'mysql+pdo://user1:pass1@host1:10001/db',
```

Comment:
parameters from dsn should overwrite ones set in the configuration.

Reply:
Done. Due that dbal ignores other connection settings if `url` is set we simply just set the fq `url` and we are done.